test/unit/mock/fake_runtime_client.py

```diff
@@ -117,7 +117,7 @@ def __init__(
     ):
         """Initialize a fake job."""
         self._job_id = job_id
-        self._status = final_status or "QUEUED"
+        self._state = {"status": final_status or "QUEUED"}
```

Comment:
Why do you set `self._state` in the fake job and other instance variables in the "real" job?
```
self._status = JobStatus.INITIALIZING
self._reason: Optional[str] = None
```

Reply:
Shouldn't it be in sync?